src/engine/schema.rs

```diff
@@ -1865,21 +1938,144 @@ fn generate_static_version_query() -> Property {
+        PropertyKind::Scalar,
+        "String".to_string(),
+    )
+}
+
+fn string_array_input(name: &str) -> Property {
+    Property {
+        name: name.to_string(),
+        kind: PropertyKind::Scalar,
+        type_name: "String".to_string(),
+        required: false,
+        list: true,
+        arguments: HashMap::new(),
+        resolver: None,
+        validator: None,
+    }
+}
+
+fn fmt_int_query_input_name() -> String {
+    "IntQueryInput".to_string()
+}
+
+fn int_query_input() -> NodeType {
+    NodeType::new(
+        fmt_int_query_input_name(), 
+        TypeKind::Input, 
+        hashmap! {
+            "EQ".to_string() => int_input("EQ"),
+            "NOTEQ".to_string() => int_input("NOTEQ"),
+            "IN".to_string() => int_input("IN"),
+            "NOTIN".to_string() => int_input("NOTIN"),
+            "GT".to_string() => int_input("GT"),
+            "GTE".to_string() => int_input("GTE"),
+            "LT".to_string() => int_input("LT"),
+            "LTE".to_string() => int_input("LTE"),
+        }
+    )
+}
+
+fn int_input(name: &str) -> Property {
+    Property::new(
+        name.to_string(),
+        PropertyKind::Scalar,
+        "Int".to_string(),
+    )
+
+}
+
+fn fmt_float_query_input_name() -> String {
+    "FloatQueryInput".to_string()
+}
+
+fn float_query_input() -> NodeType {
+    NodeType::new(
+        fmt_int_query_input_name(), 
+        TypeKind::Input, 
+        hashmap! {
+            "EQ".to_string() => float_input("EQ"),
+            "NOTEQ".to_string() => float_input("NOTEQ"),
+            "IN".to_string() => float_input("IN"),
+            "NOTIN".to_string() => float_input("NOTIN"),
+            "GT".to_string() => float_input("GT"),
+            "GTE".to_string() => float_input("GTE"),
+            "LT".to_string() => float_input("LT"),
+            "LTE".to_string() => float_input("LTE"),
+        }
+    )
+}
+
+fn float_input(name: &str) -> Property {
+    Property::new(
+        name.to_string(),
+        PropertyKind::Scalar,
+        "Float".to_string(),
+    )
+}
+
 /// Takes a WG config and returns a map of graphql schema components for model
 /// types, custom endpoints, and associated endpoint types
-fn generate_schema(c: &Configuration) -> HashMap<String, NodeType> {
+fn generate_schema(c: &Configuration) -> Result<HashMap<String, NodeType>, Error> {
     let mut nthm = HashMap::new();
     let mut mutation_props = HashMap::new();
     let mut query_props = HashMap::new();
 
+    // StringQueryInput
+    nthm.insert(
+        fmt_string_query_input_name(),
+        string_query_input()
+    );
+
+    // NumberQueryInput
+    nthm.insert(
+        fmt_int_query_input_name(),
+        int_query_input()
+    );
+    
+    // FloatQueryInput
+    nthm.insert(
+        fmt_float_query_input_name(),
+        float_query_input()
+    );
+
     // generate graphql schema components for warpgrapher types
-    c.types().for_each(|t| {
+    //c.types().for_each(|t| {
```

Comment:
I have no complaint about switching to a for loop, if you prefer it, but there is an Iterator method try_for_each that would allow you to error chain with `?` in the closure, if that's why you made the switch.

Reply:
In general when I'm mutating external state (outside the scope of the block) I prefer to use for loops - but that's a personal style preference. No objection to reverting if you prefer.

Reply:
Nah, it's fine by me.